blues/models.py

```diff
@@ -0,0 +1,317 @@
+        system defined by particles1-3
+    '''
+    #calculate new basis set
+    vec1, vec2, vec3 = localcoord(particle1, particle2, particle3)
+    basis_set = np.zeros((3,3))*unit.nanometers
+    basis_set[0] = vec1
+    basis_set[1] = vec2
+    basis_set[2] = vec3
+    #since the origin is centered at particle1 by convention
+    #subtract to account for this
+    recenter = center - particle1
+    #find coordinate in new coordinate system
+    new_coord = changeBasis(basis_set, recenter)
+    old_coord = undoBasis(basis_set, new_coord)
+    return new_coord
+
+def findOldCoord(particle1, particle2, particle3, center):
+    '''Finds the coordinates of a given center (defined by a different basis
+    given by particles1-3) back in euclidian coordinates
+        system defined by particles1-3
+    '''
+
+    vec1, vec2, vec3 = localcoord(particle1, particle2, particle3)
+    basis_set = np.zeros((3,3))*unit.nanometers
+    basis_set[0] = vec1
+    basis_set[1] = vec2
+    basis_set[2] = vec3
+    #since the origin is centered at particle1 by convention
+    #subtract to account for this
+    old_coord = undoBasis(basis_set, center)
+    adjusted_center = old_coord + particle1
+    return adjusted_center
+
+class Model_SmartDart(Model):
+    def __init__(self, structure, basis_particles, dart_size=0.2*unit.nanometers, resname='LIG'):
+        super(Model_SmartDart, self).__init__(structure, resname=resname)
+        self.dartboard = []
+        self.n_dartboard = []
+        self.particle_pairs = []
+        self.particle_weights = []
+        self.basis_particles = basis_particles
+        self.dart_size = dart_size
+        self.calculateProperties()
+
+    def dartsFromParmEd(self, coord_files, topology=None):
+        """
+        Used to setup darts from a generic coordinate file, through MDtraj using the basis_particles to define
+        new basis vectors, which allows dart centers to remain consistant through a simulation.
+        This adds to the self.n_dartboard, which defines the centers used for smart darting.
+        Arguments
+        ---------
+        system: simtk.openmm.system
+            Openmm System corresponding to the system to smart dart.
+        coord_files: list of str
+            List containing coordinate files of the system for smart darting.
+        atom_indices: list of ints
+            List containing the ligand atom indices. If None uses self.atom_indices instead.
+        basis_particles: list of 3 ints
+            Specifies the 3 indices of particles whose coordinates will be used
+            as basis vectors. If None is specified, uses those found in basis particles.
+            If None uses self.basis_particles instead.
+        Returns
+        -------
+        n_dartboard: list of 1x3 np.arrays
+            Center of mass coordinates of atom_indices particles in new basis set.
+        """
+        atom_indices = self.atom_indices
```

Comment:
It's a class method, and I can't think of a typical case where you'd want to use anything other than the model.atom_indicies so I don't see a particular reason to change it. I did leave some leftover docstring in there though, which I can get rid of.